packages/aml-checklist/src/sagas/api.ts

```diff
@@ -0,0 +1,159 @@
+    logger(err)
+    return err?.response
+  }
+}
+
+export const fetchIdentityCheck = async ({ contactId, headers }) => {
+  try {
+    const response = await fetcher({
+      url: `${URLS.idChecks}?ContactId=${contactId}`,
+      api: window.reapit.config.platformApiUrl,
+      method: 'GET',
+      headers: headers,
+    })
+    const newResponse = {
+      ...response,
+      _embedded: response?._embedded.map(identityCheck => {
+        return changeTimeZoneLocalForIdentityCheck(identityCheck)
+      }),
+    }
+    return newResponse?._embedded?.[0] || null
+  } catch (err) {
+    logger(err)
+    return err?.response
+  }
+}
+
+export const updateChecklist = async ({ contact, headers }) => {
+  try {
+    const formattedContact = formatDateForContact(contact)
+    const { _eTag, ...otherData } = formattedContact
+    const response = await fetcher({
+      url: `${URLS.contacts}/${contact.id}`,
+      api: window.reapit.config.platformApiUrl,
+      method: 'PATCH',
+      headers: {
+        ...headers,
+        'If-Match': _eTag,
+      },
+      body: otherData,
+    })
+    return response
+  } catch (err) {
+    logger(err)
+    return err?.response
+  }
+}
+
+export const uploadImage = async ({ name, imageData, headers }) => {
+  // The fileuploader is not a Platform API. As such, api version header will throw if not deleted
+  const headersWithoutVersion = {
+    ...headers,
+  }
+  delete headersWithoutVersion['api-version']
+  try {
+    const response = await fetcher({
+      url: '/',
+      api: window.reapit.config.uploadApiUrl,
+      method: 'POST',
+      headers: headersWithoutVersion,
+      body: {
+        name,
+        imageData,
+      },
+    })
+    return response
+  } catch (err) {
+    logger(err)
+    return err?.response
+  }
+}
+
+export const updateIdentityCheck = async ({ identityChecks, headers }) => {
+  try {
+    const { _eTag, ...otherData } = identityChecks
+    const formatedIdentityChecks = changeTimeZoneUTCForIdentityCheck(otherData)
+    const response = await fetcher({
+      url: `${URLS.idChecks}/${identityChecks.id}`,
+      api: window.reapit.config.platformApiUrl,
+      method: 'PATCH',
+      headers: {
+        ...headers,
+        'If-Match': _eTag,
+      },
+      body: formatedIdentityChecks,
+    })
+    return response
+  } catch (err) {
+    logger(err)
+    return err?.response
+  }
+}
+
+export const createIdentityCheck = async ({ identityChecks, headers }) => {
+  try {
+    const formatedIdentityChecks = changeTimeZoneUTCForIdentityCheck(identityChecks)
+    const response = await fetcher({
+      url: URLS.idChecks,
+      api: window.reapit.config.platformApiUrl,
+      method: 'POST',
+      headers: headers,
+      body: formatedIdentityChecks,
+    })
+    return response
+  } catch (err) {
+    logger(err)
+    return err?.response
+  }
+}
+
+export const fetchContact = async ({ params, headers }) => {
+  try {
+    const responseContacts = await fetcher({
```

Comment:
I think you should use `cleanObject` to remove all empty values first before request, only this API need that, all other apis are fine
https://github.com/reapit/foundations/blob/master/packages/aml-checklist/src/sagas/result.ts#L14-L24

Reply:
fixed